### DISCUSSION TITLE
Enhancement: Midi Input Mapping

### DISCUSSION THREAD

**Discussion by RaincloudTheDragon** [FunkinCrew/Funkin#1643] [Ideas]
Now that the version in Week 7 has the ability to change the controls (along with a whole host of quality-of-life improvements which I consider to be a fantastic step in the right direction), perhaps native midi controller support could be implemented as well? I currently use a program called Midi2Keys, and then remap controls so my Midi Fighter 3d can play the notes, which is adequate, but requires more steps and other programs to run. I've heard of other rhythm games having native midi support, as in changing your inputs to different keybinds could also be changed to midi keys in the exact same way. Would be a great and fairly easy feature to integrate, albeit not a major one.

**Comment by aflacc** [comment]
This would be cool. I feel like the debug menu needs an overhaul. What if we had a section or a feature that allows us to implement poses like the "Hey" or "Fear" by pressing a key?

I also feel like we should be able to play the song at the point of the cursor. Maybe Shift+Enter would cause you to start the song at the cursor or maybe even just the currently selected section. We also should have a recording feature where if you press space to preview the song maybe pressing arrow keys or wasd or something could cause notes to appear at the specified spot. Holding the key down would place a hold note
